.github/workflows/test.yml

```diff
@@ -0,0 +1,59 @@
+name: Test tap-postgres
+
+on: [push]
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    services:
+      # Label used to access the service container
+      postgres:
+        # Docker Hub image
+        image: postgres
+        # Provide the password for postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+        ports:
+          - 5432:5432
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    env:
+      PIP_CONSTRAINT: .github/workflows/constraints.txt
+    strategy:
+      matrix:
+        include:
+        - {python-version: "3.10", session: "lint", os: "ubuntu-latest" }
+        - {python-version: "3.10", session: "pytest", os: "ubuntu-latest" }
+        - {python-version: "3.9", session: "pytest", os: "ubuntu-latest" }
+        - {python-version: "3.8", session: "pytest", os: "ubuntu-latest" }
+        - {python-version: "3.7", session: "pytest", os: "ubuntu-latest" }
+
+    steps:
+
+    - name: Checkout code
+      uses: actions/checkout@v3.3.0
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4.5.0
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Upgrade pip
+      run: |
+        pip install pip
+        pip --version
+    - name: Install Poetry
+      run: |
+        pipx install poetry
+        poetry --version
+    - name: Install tox
+      run: |
+        pipx install tox
```

Comment:
I normally just go `poetry run tox -e lint` but this works too. 

I think I like `poetry` more as I keeps the version stuck to `pyproject.yaml` and will remove the `tox` need in constaints.txt. Can swap to that? 

Take a look at `target-postgres` https://github.com/MeltanoLabs/target-postgres/blob/main/.github/workflows/ci_workflow.yml

Reply:
I think I resolved this the way you wanted me to but lmk if I misunderstood or made a mistake when doing that switch.